### DISCUSSION TITLE
How can I force DfaMatcher to log an error when no candidates are found?

### DISCUSSION THREAD

**Discussion by DefinitelyADev** [dotnet/yarp#2202] [Q&A] [answered]
Hello all!

Is the a way to capture, in order to log as error, the case "No candidates found for the request path..."?
I am thinking of switching `Microsoft.AspNetCore.Routing.Matching.DfaMatcher` to debug but, I don't see the point of having "x candidate(s) found for the request path..." when the request is correct. I just want to log (and maybe add a prometheus metric), for the paths with no candidates.

Thanks!

**Comment by Tratcher** [comment, marked as answer]
You could log in middleware if no Endpoint is set. (HttpContext.GetEndpoint() == null).

**Reply by Tratcher** [reply]
In the future https://github.com/dotnet/runtime/issues/82465 could help here.
